.vsts-dotnet.yml

```diff
@@ -13,7 +13,8 @@ trigger:
 variables:
   - name: SourceBranch
     value: $(IbcSourceBranchName)
-  - ${{ if startsWith(variables['Build.SourceBranch'], 'refs/heads/exp/') }}:
+  - ${{ if and(eq(variables['Build.SourceBranch'], variables['IbcSourceBranchName']), startsWith(variables['Build.SourceBranch'], 'refs/heads/exp/')) }}:
```

Comment:
I don't think I follow.

Wouldn't this if statement always be true in an `exp/` branch? Just above this, `SourceBranch` is set to `IbcSourceBranchName`, then we check if it's an `exp/` branch and it's `SourceBranch` matches the `IbcSourceBranchName`, then we set `SourceBranch` to master. 

How does this allow overriding?

Reply:
`Build.SourceBranch` and `SourceBranch` are distinct: `SourceBranch` is a local concept and `Build.SourceBranch` is assigned by the system to be the real requested branch name.

In pseudocode:

```
SourceBranch = IbcSourceBranchName
if (realSourceBranch == IbcSourceBranchName &&
    realSourceBranch.startsWith('refs/heads/exp/`):
        SourceBranch = 'master'
```

Reply:
Got it, so this allows us to use other optprof branches to workaround master's optprof branch being out of whack.